frontend/BoardGUI.py

```diff
@@ -33,26 +35,45 @@ def draw_piece(self, piece, position):
         else:
             self.game.screen.blit(self.white_piece_image, position)
 
+    # def get_cell_clicked(self):
+    #     current_cell = vec2(pg.mouse.get_pos()) // self.cell_size
+    #     col, row = map(int, current_cell)
+    #     left_click = pg.mouse.get_pressed()[0]
+    #     if left_click:
+    #         # print(self.backend_game.get_board(row,col), self.count)
+    #         self.count += 1
+
+    #         if not self.backend_game.placement_complete():
+    #             x = self.backend_game.piece_placement(row, col)
+    #             # print(x, "\n")
+    #             if x == 1:
+    #                 self.draw_piece(self.backend_game.get_turn(),current_cell * self.cell_size)
+    #                 self.backend_game.end_turn()
+    #         else:
+    #             print("Movement")
+    #     # if left_click and self.board[row][col].state == CellType.EMPTY:
+    #         # self.board.board[row][col].setState(CellType.BLACK)
+    #     #     print(row, col)
+    #     #     print(self.board.board[row][col].state)
+
+    #     return current_cell
+    #     # return col, row
     def get_cell_clicked(self):
         current_cell = vec2(pg.mouse.get_pos()) // self.cell_size
         col, row = map(int, current_cell)
-        left_click = pg.mouse.get_pressed()[0]
-        if left_click:
-            # print(self.backend_game.get_board(row,col), self.count)
-            # self.count += 1
+        mouse_buttons = pg.mouse.get_pressed()  # Call this once and store the result
 
+        if mouse_buttons[0]:  # Left click
+            self.count += 1
             if not self.backend_game.placement_complete():
```

Comment:
I believe in the next sprint we should put this into the GameManager class so that the BoardGUI isnt controlling the turn switching. I think it would be okay if the GameManager class draws the pieces onto the board or we could store the turn in a variable that we send back to the BoardGUI to keep it separate. We can discuss this more later.